numpy/random/meson.build

```diff
@@ -66,7 +66,7 @@ random_pyx_sources = [
       fs.copyfile('mtrand.pyx'),
       'src/distributions/distributions.c',
       'src/legacy/legacy-distributions.c'
-    ], ['-DNPY_RANDOM_LEGACY=1'], npymath_lib,
+    ], ['-DNP_RANDOM_LEGACY=1'], npymath_lib,
```

Comment:
This typo took wayyyy too long to find. The build did not crash locally, only on CI, and it took quite a bit of effort to get meson to emit pdb files that I could download and use locally.

Reply:
Maybe we should define `NPY_RANDOM_LEGACY` as an alias. `NPY` looks like the expected prefix.

Reply:
xref the meson tracking issue #22546

Reply:
It is legacy so I don't think we should change/alias it. But it sure was annoying to find. I wonder if there is a way we could define it inside mtrand.pyx instead, but that is beyond this PR.

Reply:
ugh yes, this is ugly and that was an unfortunate typo.

Reply:
removing it from the build definition files sound like a nice follow-up. I think I agree with not adding the `NPY_` flavor.